signer/tests/integration/transaction_coordinator.rs

```diff
@@ -325,3 +331,111 @@ async fn process_complete_deposit() {
 
     testing::storage::drop_db(db).await;
 }
+
+/// The [`TxCoordinatorEventLoop::get_signer_set_and_aggregate_key`]
+/// function is supposed to fetch the "current" signing set and the
+/// aggregate key to use for bitcoin transactions. It attempts to get the
+/// latest rotate-keys contract call transaction confirmed on the canonical
+/// Stacks blockchain and falls back to the DKG shares table if no such
+/// transaction can be found.
+///
+/// This tests that we prefer rotate keys transactions if it's available
+/// but will use the DKG shares behavior is indeed the case.
+#[cfg_attr(not(feature = "integration-tests"), ignore)]
+#[tokio::test]
+async fn get_signer_public_keys_and_aggregate_key_falls_back() {
+    let db_num = DATABASE_NUM.fetch_add(1, Ordering::SeqCst);
+    let db = testing::storage::new_test_database(db_num, true).await;
+
+    let mut rng = rand::rngs::StdRng::seed_from_u64(51);
+
+    let ctx = TestContext::builder()
+        .with_storage(db.clone())
+        .with_mocked_clients()
+        .build();
+
+    let network = InMemoryNetwork::new();
+
+    let coord = TxCoordinatorEventLoop {
+        network: network.connect(),
+        context: ctx.clone(),
+        context_window: 10000,
+        private_key: ctx.config().signer.private_key,
+        signing_round_max_duration: Duration::from_secs(10),
+        threshold: 2,
+    };
+
+    // We need stacks blocks for the rotate-keys transactions.
+    let test_params = testing::storage::model::Params {
+        num_bitcoin_blocks: 10,
+        num_stacks_blocks_per_bitcoin_block: 1,
+        num_deposit_requests_per_block: 0,
+        num_withdraw_requests_per_block: 0,
+        num_signers_per_request: 0,
+    };
+    let test_data = TestData::generate(&mut rng, &[], &test_params);
+    test_data.write_to(&db).await;
+
+    // We always need the chain tip.
+    let chain_tip = db.get_bitcoin_canonical_chain_tip().await.unwrap().unwrap();
+
+    // We have no rows in the DKG shares table and no rotate-keys
+    // transactions, so this should error
+    let ans = coord.get_signer_set_and_aggregate_key(&chain_tip).await;
+    assert!(ans.is_err());
+
+    // Alright, lets write some DKG shares into the database. When we do
+    // that the signer set should be considered whatever the signer set is
+    // from our DKG shares.
+    let shares: EncryptedDkgShares = Faker.fake_with_rng(&mut rng);
+    db.write_encrypted_dkg_shares(&shares).await.unwrap();
+
+    let (aggregate_key, signer_set) = coord
+        .get_signer_set_and_aggregate_key(&chain_tip)
+        .await
+        .unwrap();
+
+    let shares_signer_set: BTreeSet<PublicKey> =
+        shares.signer_set_public_keys.iter().copied().collect();
+
+    assert_eq!(shares.aggregate_key, aggregate_key);
+    assert_eq!(shares_signer_set, signer_set);
+
+    // Okay not we write a rotate-keys transaction into the database. To do
```

Comment:
```suggestion
    // Okay now we write a rotate-keys transaction into the database. To do
```

Reply:
Updated in https://github.com/stacks-network/sbtc/pull/655/commits/0362a3eb90a0ea06d227866197ddec65e6e092d6